pkg/tracking/store/sql/tags.go

```diff
@@ -80,3 +81,182 @@ func (s TrackingSQLStore) setTagsWithTransaction(
 
 	return nil
 }
+
+const (
+	maxEntityKeyLength = 250
+	maxTagValueLength  = 8000
+)
+
+// Helper function to validate the tag key and value
+func validateTag(key, value string) *contract.Error {
```

Comment:
This should not at all happen here.
The validation code happens in the `generate/validations.go`.
You have put `required` in there, this will check for empty strings.

Reply:
When I wrote this code, I was trying to fix the failing tests which seemed related to the validation. Upon investigating further, the test result doesn't change when removing the `validateTag` function.